File: src/main/java/aurocosh/divinefavor/common/potions/potions/PotionLiquidWalking.java
package aurocosh.divinefavor.common.potions.potions;

import aurocosh.divinefavor.common.potions.base.potion.ModPotion;
import net.minecraft.block.Block;
import net.minecraft.entity.EntityLivingBase;

import static aurocosh.divinefavor.common.util.UtilEntity.tickLiquidWalk;

public class PotionLiquidWalking extends ModPotion {
    private final Block block;

    public PotionLiquidWalking(String name, Block block) {
        super(name, true, 0x7FB8A4);
        this.block = block;
    }

    @Override
    public void performEffect(EntityLivingBase livingBase, int amplifier) {
        tickLiquidWalk(livingBase, block);
    }

    @Override
    public boolean isReady(int duration, int amplifier) {
        return true;
    }
}

File: src/main/java/aurocosh/divinefavor/common/potions/potions/PotionLiquidWalking.kt
package aurocosh.divinefavor.common.potions.potions

import aurocosh.divinefavor.common.potions.base.potion.ModPotion
import aurocosh.divinefavor.common.util.UtilEntity.tickLiquidWalk
import net.minecraft.block.Block
import net.minecraft.entity.EntityLivingBase

class PotionLiquidWalking(name: String, private val block: Block) : ModPotion(name, true, 0x7FB8A4) {

    override fun performEffect(livingBase: EntityLivingBase, amplifier: Int) {
        tickLiquidWalk(livingBase, block)
    }

    override fun isReady(duration: Int, amplifier: Int): Boolean {
        return true
    }
}
